### DISCUSSION TITLE
Rotate

### DISCUSSION THREAD

**Discussion by RCAFinfo** [PintaProject/Pinta#1135] [Ideas]
OK, brand new user here and a loyal user of paint.net. Attempting to move to Linux and gimp as a super click heavy application that requires so many extra steps to do simple operations.
Tried this application starting about half an hour ago and it is faster than paint.net. There are 2 quick suggestions that I have.
1) is there a way for you to modify the rotate / zoom layer to allow for a finer rotate (for example 0.25 degrees) and allow for a negative rotate? To go with this it would be fabulous to have the canvas resize to the outer edges of the rotates image.
2) is there a way to implement a default tool selected on opening an new image?

I am totally impressed with this application so far and this will definitely ease my transition to Linux, thank you.

**Comment by cameronwhite** [comment]
- I think you should be able to type in fractional values in the Rotate/Zoom dialog, it's just the +/- buttons which increment by a single degree. Negative rotation would be good to add though - feel free to submit that to https://github.com/PintaProject/Pinta/issues
- Pinta should remember what your last used tool was when it's reopened, or do you mean always immediately switching to a tool whenever a new image is opened up while Pinta's already open?

**Reply by RCAFinfo** [reply]
I added the negative rotation enhancement to the issue to the issues.

- I attempted to do a fractional/ decimal rotation and the entry box only appears to allow integers. the "." and "/" characters are not allowed. 
- when you do a rotate in the rotate/zoom dialog the canvas size remains static cropping off any area of the image that now exits the canvas size, it would be awesome if the canvas size enlarged according to the new dimensions of the original image.
- Pinta does remember the last tool that was used which works good for some work flows. most of the work that I do runs a very similar set of steps on each scanned image that I am processing. If I could have the Rectangle select tool activated every time that I open an image it would make my life much easier. I know this is a picky one. My thought on implementing this suggestion would be to implement a preference / options screen and have a drop down box with the various tools and a default option to retain the last tool used (as it is currently) I do not know how viable this is.

**Reply by cameronwhite** [reply]
Ah, it seems like the fractional rotation is working the `master` branch, so this must have changed at some point while porting Pinta to GTK4. Testing Pinta 2.1 it indeed only accepts integers

The other two I think can also be submitted as issues! The preference needs some more thought since we generally aim for simplicity vs having big configuration dialogs, so I'd probably lean more towards e.g. right clicking on the tool to configure that preference

**Reply by RCAFinfo** [reply]
Done, Thank you for your input and responsiveness.

**Comment by ryan-de-boer** [comment]
I ran into the problem that the canvas does not resize on rotate. Have to quickly dip out to another program for rotation for now. Thanks for the post. I found the issue here in case someone is searching: https://github.com/PintaProject/Pinta/issues/1145
Wow paint .net and Krita have the same problem! I found Affinity works though, so at least some free program works.

**Comment by ryan-de-boer** [comment]
I found how it works in Pinta. Do Image/Rotate 90 degrees clockwise/counter-clockwise, and then it rotates the canvas. Image is that hard to find icon in the top right next to view.
I suppose the other section is for rotating layers, which you may not want to rotate the canvas? Not sure. Anyway the rotate 90 degrees is my use case so I no longer need to switch to Affinity.
